WcaOnRails/app/models/competition.rb

```diff
@@ -197,6 +197,41 @@ class Competition < ApplicationRecord
     end
   end
 
+  # Only validate on update: nobody can confirm competition on creation.
+  # The only exception to this is within tests, in which case we actually don't want to run this validation.
+  validate :schedule_must_match_rounds, if: :confirmed_at_changed?, on: :update
```

Comment:
The schedule can change after a competition gets confirmed, why would we validate it only when `confirmed_at` changes?

Reply:
Hum weird that I didn't include a comment about this.
Events and schedule are only editable separately, if we validate `schedule_must_match_rounds` always we'll have two issues:
  - old competitions will be invalid
  - but most importantly, changing the number of rounds for announced competition won't be possible: adding rounds through the events page won't be possible because the schedule won't contain the round just added.

Until we have a unified events/schedule page, I don't think we can validate this every time.

Maybe you have a working idea to deal with that?

Reply:
Ah thanks, that makes sense, it definitely deserves a comment =)
As you said, probably no easy way of handling this without a single page for events and schedule, we can think it through later on.